packages/components/src/spectrum/CheckboxGroup.tsx

```diff
@@ -18,38 +20,24 @@ export function CheckboxGroup({
   children,
   ...props
 }: CheckboxGroupProps): JSX.Element {
-  const [checkedState, setCheckedState] = useState<{ [key: number]: boolean }>(
-    {}
-  );
-
-  const handleCheckboxChange = (index: number) => {
-    setCheckedState(prevState => ({
-      ...prevState,
-      [index]: !prevState[index],
-    }));
-  };
-
   const wrappedChildren = useMemo(
     () =>
-      React.Children.map(children, (child, index) => {
-        if (isElementOfType(child, Checkbox)) {
-          return React.cloneElement(child, {
-            isSelected: true,
-            value: `checkbox-${index}`,
-            onChange: () => handleCheckboxChange(index),
-          });
-        }
-        return (
+      ensureArray(children).map((child, index) =>
+        isElementOfType(child, Checkbox) ? (
+          React.cloneElement(child, {
+            key: `${index}-${String(child)}`,
+            value: `${index}-${String(child)}`,
```

Comment:
Why does `value` include `index`?

Reply:
Actually, we shouldn't be cloning here either. <Checkbox/> children passed in need to provide their own values (see Spectrum examples)